frontend/model/contracts/group.js

```diff
@@ -256,6 +256,30 @@ DefineContract({
         archiveProposal(state, data.proposalHash)
       }
     },
+    'gi.contracts/group/removeMember': {
+      validate: objectOf({
+        username: string
+      }),
+      process (state, { data, meta }) {
+        const rootState = sbp('state/vuex/state')
+        if (data.username === rootState.loggedIn.username) {
+          console.warn('TODO - Im the member being removed... what happens?')
+          // Should remove contracts.
+          // Set currentGroupId to null or change to another group? Show any info about it?
+        } else {
+          Vue.delete(state.profiles, data.username)
+        }
+      },
+      async sideEffect (message) {
+        const rootState = sbp('state/vuex/state')
+        const data = message.data()
+        if (data.username === rootState.loggedIn.username) {
+          console.warn('TODO sideEffect - Remove all related contracts. How?')
+        } else {
+          console.warn('TODO sideEffect - Remove member from state', data.username)
+        }
```

Comment:
@taoeffect a little guidance from you about what's your vision for a possible `enqueueRemoveContractSync` would be helpful here :)


Another scenario I remember: John and Bob have 2 different groups in common. Bob is removed from one of the groups. In that case, we can't remove Bob's `identityContract` from John's state because he would still need it on the other group. What's the best approach to detect/avoid this conflict?